profile-widget/profilewidget2.cpp

```diff
@@ -1399,6 +1399,26 @@ struct int ProfileWidget2::getEntryFromPos(QPointF pos)
 #endif
 
 #ifndef SUBSURFACE_MOBILE
+static QString makeCylinderDescription(int i, cylinder_t *cylinder)
+{
+	// TODO: Figure out how to translate strings here, kindof messy adding it to the class.
+	//QString mix = tr("air");
+	QString mix = "air";
+	int o2 = cylinder->gasmix.o2.permille / 10;
+	int he = cylinder->gasmix.he.permille / 10;
+	if (o2 > 0) {
+		if(he > 0) {
+			mix = QString("%1/%2").arg(o2,he);
+		} else {
+			mix = QString("EAN%1").arg(o2);
+		}
+	}
```

Comment:
instead of figuring this all out here, how about you use `get_gas_string()` from `qthelper.cpp` -- that one already deals with creating consistent names, with translations, etc.

Reply:
Even for calculations you should not access members of a struct gasmix directly but use the get_o2 and get_he helpers as they treat the special case of air correctly.

Reply:
Thank you. I _did_ try searching for "EAN" because I was sure there'd be a helper function around for it, but couldn't find it. I probably did a file-local search instead of project-wide in QtCreator. In future I'll grep. ;)

Committed the changes and squashed all the commits into one, as requested, as well as keeping to the 74-char line-limit in the commit message.